### DISCUSSION TITLE
vulnerability report

### DISCUSSION THREAD

**Discussion by luisfilsilva** [artilleryio/artillery#1229] [Artillery]
Hi experts,

When I install artillery, I get this vulnerability report:

                                                            
                       === npm audit security report ===                        
                                                                                
┌──────────────────────────────────────────────────────────────────────────────┐
│                                Manual Review                                 │
│            Some vulnerabilities require your attention to resolve            │
│                                                                              │
│         Visit https://go.npm.me/audit-guide for additional guidance          │
└──────────────────────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │  Inefficient Regular Expression Complexity in                │
│               │ chalk/ansi-regex                                             │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ ansi-regex                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=5.0.1                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ artillery                                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ artillery > @oclif/plugin-help > wrap-ansi > strip-ansi >    │
│               │ ansi-regex                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://github.com/advisories/GHSA-93q8-gq69-wqmw            │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │  Inefficient Regular Expression Complexity in                │
│               │ chalk/ansi-regex                                             │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ ansi-regex                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=5.0.1                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ artillery                                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ artillery > @oclif/plugin-help > wrap-ansi > string-width >  │
│               │ strip-ansi > ansi-regex                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://github.com/advisories/GHSA-93q8-gq69-wqmw            │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │  Inefficient Regular Expression Complexity in                │
│               │ chalk/ansi-regex                                             │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ ansi-regex                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=5.0.1                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ artillery                                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ artillery > @oclif/command > @oclif/plugin-help > wrap-ansi  │
│               │ > string-width > strip-ansi > ansi-regex                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://github.com/advisories/GHSA-93q8-gq69-wqmw            │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │  Inefficient Regular Expression Complexity in                │
│               │ chalk/ansi-regex                                             │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ ansi-regex                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=5.0.1                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ artillery                                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ artillery > cli-table3 > string-width > strip-ansi >         │
│               │ ansi-regex                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://github.com/advisories/GHSA-93q8-gq69-wqmw            │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Prototype Pollution in dot-prop                              │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ dot-prop                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=4.2.1                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ artillery                                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ artillery > sketches-js-hassy > math-float64-frexp >         │
│               │ math-float64-from-words > utils-is-little-endian >           │
│               │ update-notifier > configstore > dot-prop                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://github.com/advisories/GHSA-ff7x-qrg7-qggm            │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Prototype Pollution in dot-prop                              │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ dot-prop                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=4.2.1                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ artillery                                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ artillery > sketches-js-hassy > math-float64-frexp >         │
│               │ math-float64-exponent > math-float64-get-high-word >         │
│               │ utils-is-little-endian > update-notifier > configstore >     │
│               │ dot-prop                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://github.com/advisories/GHSA-ff7x-qrg7-qggm            │
└───────────────┴──────────────────────────────────────────────────────────────┘
found 6 vulnerabilities (4 moderate, 2 high) in 572 scanned packages
  6 vulnerabilities require manual review. See the full report for details.

It seems that some libraries are using dot-prop and ansi-regex versions with some vulnerabilities. Can we overwrite these libraries and use the latest versions?

Thanks,
Luís

**Comment by hassy** [comment]
Will take a look, thanks @luisfilsilva. In general, the motivation behind `npm audit` was great, but the way it's implemented its warnings can be ignored 99% of the time (as in this case) - https://www.theregister.com/2021/07/09/npm_audit_security/

**Reply by luisfilsilva** [reply]
Thanks @hassy! Will take that into account.
